src/main/java/frc/robot/subsystems/endEffector/EndEffector.java

```diff
@@ -0,0 +1,66 @@
+package frc.robot.subsystems.endEffector;
+
+import frc.robot.hardware.digitalinput.DigitalInputInputsAutoLogged;
+import frc.robot.hardware.digitalinput.IDigitalInput;
+import frc.robot.hardware.interfaces.ControllableMotor;
+import frc.robot.hardware.interfaces.IRequest;
+import frc.robot.subsystems.GBSubsystem;
+
+public class EndEffector extends GBSubsystem {
+
+	private final ControllableMotor roller;
+	private final IRequest<Double> powerRequest;
+	private final IDigitalInput frontBeamBreaker;
+	private final DigitalInputInputsAutoLogged frontBeamBreakerInputs;
+	private final IDigitalInput backBeamBreaker;
+	private final DigitalInputInputsAutoLogged backBeamBreakerInputs;
+	private final EndEffectorCommandsBuilder commandsBuilder;
+
+	public EndEffector(
+		ControllableMotor roller,
+		IDigitalInput frontBeamBreaker,
+		IDigitalInput backBeamBreaker,
+		IRequest<Double> powerRequest,
+		String logPath
+	) {
+		super(logPath);
+		this.roller = roller;
+		this.powerRequest = powerRequest;
+		this.frontBeamBreaker = frontBeamBreaker;
+		this.frontBeamBreakerInputs = new DigitalInputInputsAutoLogged();
+		this.backBeamBreaker = backBeamBreaker;
+		this.backBeamBreakerInputs = new DigitalInputInputsAutoLogged();
+		this.commandsBuilder = new EndEffectorCommandsBuilder(this);
+	}
+
+	public boolean isCoralInFrontBeamBreaker() {
```

Comment:
consider changing the name to something simpler:
```suggestion
	public boolean isCoralInFront() {
```